flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -844,14 +844,16 @@ private ApplicationReport startAppMaster(
 		tmpConfigurationFile.deleteOnExit();
 		BootstrapTools.writeConfiguration(configuration, tmpConfigurationFile);
 
+		String flinkConfigKey = "flink-conf.yaml";
 		Path remotePathConf = setupSingleLocalResource(
-				"flink-conf.yaml",
+			flinkConfigKey,
 				fs,
 				appId,
 				new Path(tmpConfigurationFile.getAbsolutePath()),
 				localResources,
 				homeDir,
 				"");
+		envShipFileList.append(flinkConfigKey).append("=").append(remotePathConf).append(",");
```

Comment:
I'm curious this line. `envShipFileList` configures files ship on TM. With this line we actually ship `flink-conf.yaml` to TM even if we remove it from code diff above. What is your decision on whether or not ship `flink-conf.yaml` indeed?

Reply:
Yes, we always ship the `flink-conf.yaml` to all `TaskExecutor`s. The `TaskExecutor` will load flink configuration from `flink-conf.yaml` uploaded by client and dynamic properties set by `YarnResourceManager`. 

@TisonKun What do you mean "we remove it from code diff above".

Reply:
hmm. I second @TisonKun here - if we eventually have to put the `flink-conf.yaml` into the envShipFileList . wouldn't we still eventually upload the file into DFS (based on [this comment](https://github.com/apache/flink/pull/10143/files#r344503690), I thought this is the problem we were trying to avoid in the first place) ? 

(@TisonKun please correct me if I understood you wrong)

Reply:
I get the different and I think it is the fault of Utils#L493 call the tm conf file `flinkConf`. IIUC we still ship the original flink-conf.yaml, and what @wangyang0918 trying to resolve here is FLINK ship a modified flink-conf.yaml per TM launched...My concern on this issue is shared below.